### PR TITLE
Add Test View

### PR DESCRIPTION
**Pull request by graemeRharvey**

### CODE REVIEW

File: client/modules/Test/components/TestListItem/TestListItem.js [hunk 1]
```diff
@@ -4,26 +4,26 @@ import { Link } from 'react-router';
...
-function PostListItem(props) {
+function TestListItem(props) {
   return (
-    <div className={styles['single-post']}>
-      <h3 className={styles['post-title']}>
-        <Link to={`/posts/${props.post.slug}-${props.post.cuid}`} >
-          {props.post.title}
+    <div className={styles['single-test']}>
+      <h3 className={styles['test-title']}>
+        <Link to={`/tests/${props.test.slug}-${props.test.cuid}`} >
```
Comment (drewbrend) [inline client/modules/Test/components/TestListItem/TestListItem.js]: slug and cuid don't exist on a Test.

File: client/routes.js [hunk 2]
```diff
@@ -27,15 +27,15 @@ export default (
     <IndexRoute
       getComponent={(nextState, cb) => {
         require.ensure([], require => {
-          cb(null, require('./modules/Post/pages/PostListPage/PostListPage').default);
+          cb(null, require('./modules/Test/pages/TestListPage/TestListPage').default);
         });
       }}
     />
     <Route
-      path="/posts/:slug-:cuid"
```
Comment (drewbrend) [inline client/routes.js]: cuid and slug don't exist

File: client/modules/Test/components/TestListItem/TestListItem.js [hunk 3]
```diff
@@ -4,26 +4,26 @@ import { Link } from 'react-router';
...
+          {props.test.title}
         </Link>
       </h3>
-      <p className={styles['author-name']}><FormattedMessage id="by" /> {props.post.name}</p>
-      <p className={styles['post-desc']}>{props.post.content}</p>
-      <p className={styles['post-action']}><a href="#" onClick={props.onDelete}><FormattedMessage id="deletePost" /></a></p>
+      <p className={styles['author-name']}><FormattedMessage id="by" /> {props.test.name}</p>
+      <p className={styles['test-desc']}>{props.test.content}</p>
```
Comment (drewbrend) [inline client/modules/Test/components/TestListItem/TestListItem.js]: content doesn't exist

File: client/modules/Test/components/TestListItem/TestListItem.js [hunk 4]
```diff
@@ -4,26 +4,26 @@ import { Link } from 'react-router';
...
   );
 }
 
-PostListItem.propTypes = {
-  post: PropTypes.shape({
+TestListItem.propTypes = {
+  test: PropTypes.shape({
     name: PropTypes.string.isRequired,
     title: PropTypes.string.isRequired,
     content: PropTypes.string.isRequired,
```
Comment (drewbrend) [inline client/modules/Test/components/TestListItem/TestListItem.js]: wrong properties

File: client/modules/Test/pages/TestListPage/TestListPage.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+  handleDeleteTest = test => {
+    if (confirm('Do you want to delete this test')) { // eslint-disable-line
+      this.props.dispatch(deleteTestRequest(test));
+    }
+  };
+
+  handleAddTest = (name, title, content) => {
+    this.props.dispatch(toggleAddTest());
+    this.props.dispatch(addTestRequest({ name, title, content }));
```
Comment (drewbrend) [inline client/modules/Test/pages/TestListPage/TestListPage.js]: properties

File: client/modules/Test/components/TestCreateWidget/TestCreateWidget.js [hunk 6]
```diff
@@ -3,39 +3,39 @@ import PropTypes from 'prop-types';
...
       <div className={cls}>
         <div className={styles['form-content']}>
-          <h2 className={styles['form-title']}><FormattedMessage id="createNewPost" /></h2>
+          <h2 className={styles['form-title']}><FormattedMessage id="createNewTest" /></h2>
           <input placeholder={this.props.intl.messages.authorName} className={styles['form-field']} ref="name" />
-          <input placeholder={this.props.intl.messages.postTitle} className={styles['form-field']} ref="title" />
-          <textarea placeholder={this.props.intl.messages.postContent} className={styles['form-field']} ref="content" />
-          <a className={styles['post-submit-button']} href="#" onClick={this.addPost}><FormattedMessage id="submit" /></a>
+          <input placeholder={this.props.intl.messages.testTitle} className={styles['form-field']} ref="title" />
+          <textarea placeholder={this.props.intl.messages.testContent} className={styles['form-field']} ref="content" />
```
Comment (drewbrend) [inline client/modules/Test/components/TestCreateWidget/TestCreateWidget.js]: properties

File: client/modules/Test/components/TestCreateWidget/TestCreateWidget.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  };
+
+  render() {
+    const cls = `${styles.form} ${(this.props.showAddTest ? styles.appear : '')}`;
+    return (
+      <div className={cls}>
+        <div className={styles['form-content']}>
+          <h2 className={styles['form-title']}><FormattedMessage id="createNewTest" /></h2>
+          <input placeholder={this.props.intl.messages.testTitle} className={styles['form-field']} ref="name" />
```
Comment (graemeRharvey) [inline client/modules/Test/components/TestCreateWidget/TestCreateWidget.js]:
```suggestion
          <input placeholder={this.props.intl.messages.testName} className={styles['form-field']} ref="name" />
```

**Pull request merged**